cypress/e2e/models/administration/assessment_questionnaire/assessment_questionnaire.ts

```diff
@@ -125,6 +125,53 @@ export class AssessmentQuestionnaire {
                 }
             });
     }
+    public static searchQuestions(inputText: string): void {
+        cy.get(".pf-v5-c-text-input-group__text-input")
+            .type(inputText, { force: true })
+            .should("have.value", inputText);
+    }
+    static validateNumberOfMatches(section: string, expectedMatches: number): void {
+        cy.get(".pf-v5-c-tabs__item-text")
+            .contains(section)
+            .parent()
+            .find("span.pf-v5-c-badge")
+            .then(($badge) => {
+                const text = $badge.text();
+                const match = text.match(/(\d+) match(es)?/);
+                const actualMatches = match ? parseInt(match[1]) : 0;
+                expect(actualMatches).to.equal(expectedMatches);
+            });
+    }
+    static validateSearchWordInRows(textInput: string): void {
+        const lowerCaseInput = textInput.toLowerCase();
+        //loop over each row and if a row does not contain the text then click and check subtitle then close it
+        cy.get(".pf-v5-c-table > tbody > tr", { timeout: 5 * SEC })
+            .not(".pf-v5-c-table__expandable-row")
+            .each(($row) => {
+                if ($row.is(":visible")) {
+                    cy.wrap($row.find('td[data-label="Name"]'))
+                        .invoke("text")
+                        .then((cellText) => {
+                            if (!cellText.toLowerCase().includes(lowerCaseInput)) {
+                                cy.wrap($row).find("td:first").find("button").click();
+
+                                cy.wrap($row).next().invoke("is", ":visible").should("be.true");
+
+                                cy.wrap($row)
+                                    .next()
+                                    .invoke("text")
+                                    .then((expandedText) => {
+                                        expect(expandedText.toLowerCase()).to.include(
+                                            lowerCaseInput
+                                        );
+                                    });
+
+                                cy.wrap($row).find("td:first").find("button").click();
+                            }
+                        });
+                }
+            });
```

Comment:
Try this:
```
static validateSearchWordInRows(textInput: string): void {
    const lowerCaseInput = textInput.toLowerCase();

    cy.get(".pf-v5-c-table > tbody > tr:not(.pf-v5-c-table__expandable-row):visible", { timeout: 5 * SEC })
        .each(($row) => {
            const $nameColumn = $row.find('td[data-label="Name"]');
            const $expandButton = $row.find("td:first button");

            $nameColumn.invoke("text").then((cellText) => {
                if (!cellText.toLowerCase().includes(lowerCaseInput)) {
                    cy.wrap($expandButton).click();

                    cy.wrap($row.next())
                        .should("be.visible")
                        .invoke("text")
                        .then((expandedText) => {
                            expect(expandedText.toLowerCase()).to.include(lowerCaseInput);
                        });

                    cy.wrap($expandButton).click();
                }
            });
        });
}
```

Reply:
got an error "Cypress detected that you returned a promise from a command while also invoking one or more cy commands in that promise." 
But i committed an adjustment to the code as much as possible!